provider/resource_entity.go

```diff
@@ -74,6 +75,13 @@ func entityModelFromBase(ctx context.Context, b *models.Base) (m entityModel, di
 		if diags.Append(d...); diags.HasError() {
 			return
 		}
+	} else {
+		empty_set := []string{}
+		m.EntityTypeIDs, d = types.SetValueFrom(ctx, types.StringType, empty_set)
```

Comment:
this is a fix itself, framework cannot understand that it is null set of strings, even if element type is specified in schema. ```m.EntityTypeIDs = types.SetNull(types.StringType)``` works as well.

Reply:
good catch! :)